### ISSUE TITLE
idea: Implement encrypted exports

### ISSUE THREAD

**Issue by mikehelmick** [google/exposure-notifications-server#1048]
### TL;DR

For use in the importing of export files from one server to another. Implement the ability to encrypt export files so they can be produced specifically for import purposes and not distributed to clients. This will allow them to be not padded.

### Design

**Proposal**

1. Server that will be doing the importing has an RSA asymmetric key
2. Public key is shared with server that will be generating the export
3. Export server generates a key, encrypts export proto and wraps key w/ RSA public key
    . Wrapped key would be a third file in the export zip
4. Export-importer job upgraded to support unwrapping the crypto key and decrypting the export.

**Alternatives considered**

* Basic auth over the endpoint

* keep the other exports padded, and just accept fake keys will accepted into the database.

----

Just an idea at this point

**Comment by sethvargo** [comment]
I worry that encrypting the exports sends a message that implies they are "secret" in some way. We've gone to extensive lengths to convince people that the data is not PII and not a secret. In this case, encryption is basically just some additional math over a PSK, so it's really only providing the same value as a high-entropy basic-auth.

As such, my vote (in order) would be:

- Do nothing and accept that some keys are fake
- Basic auth

**Comment by mikehelmick** [comment]
At the moment it doesn't appear we need this.
/close

can always come back to it if that matters.
